test/conformance/api/v1/revision_timeout_test.go

```diff
@@ -67,37 +67,129 @@ func sendRequest(t *testing.T, clients *test.Clients, endpoint *url.URL,
+			initialSleep:   2 * time.Second,
+			expectedStatus: http.StatusOK,
+		}, {
+			name:           "exceeds timeout seconds",
+			timeoutSeconds: 10,
+			initialSleep:   12 * time.Second,
+			expectedStatus: http.StatusGatewayTimeout,
+			expectedBody:   "request timeout",
+		}, {
+			name:           "writes first byte before timeout",
+			timeoutSeconds: 10,
+			expectedStatus: http.StatusOK,
+			sleep:          15 * time.Second,
+			initialSleep:   0,
+		}}
+}
+
 func TestRevisionTimeout(t *testing.T) {
 	t.Parallel()
 	clients := test.Setup(t)
 
-	testCases := []struct {
-		name           string
-		shouldScaleTo0 bool
-		timeoutSeconds int64
-		initialSleep   time.Duration
-		sleep          time.Duration
-		expectedStatus int
-	}{{
-		name:           "does not exceed timeout seconds",
-		timeoutSeconds: 10,
-		initialSleep:   2 * time.Second,
-		expectedStatus: http.StatusOK,
-	}, {
-		name:           "exceeds timeout seconds",
-		timeoutSeconds: 10,
-		initialSleep:   12 * time.Second,
-		expectedStatus: http.StatusGatewayTimeout,
-	}, {
-		name:           "writes first byte before timeout",
-		timeoutSeconds: 10,
-		expectedStatus: http.StatusOK,
-		sleep:          15 * time.Second,
-		initialSleep:   0,
-	}}
+	for _, tc := range getDefaultTestCases() {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			names := test.ResourceNames{
+				Service: test.ObjectNameForTest(t),
+				Image:   test.Timeout,
+			}
+
+			test.EnsureTearDown(t, clients, &names)
+
+			t.Log("Creating a new Service ")
+			resources, err := v1test.CreateServiceReady(t, clients, &names, WithRevisionTimeoutSeconds(tc.timeoutSeconds))
+			if err != nil {
+				t.Fatal("Failed to create Service:", err)
+			}
+
+			serviceURL := resources.Service.Status.URL.URL()
+
+			t.Log("Probing to force at least one pod", serviceURL)
+			if _, err := pkgtest.CheckEndpointState(
+				context.Background(),
+				clients.KubeClient,
+				t.Logf,
+				serviceURL,
+				spoof.IsOneOfStatusCodes(http.StatusOK, http.StatusGatewayTimeout),
+				"CheckSuccessfulResponse",
+				test.ServingFlags.ResolvableDomain,
+				test.AddRootCAtoTransport(context.Background(), t.Logf, clients, test.ServingFlags.HTTPS)); err != nil {
+				t.Fatalf("Error probing %s: %v", serviceURL, err)
+			}
+
+			if err := sendRequest(t, clients, serviceURL, tc.initialSleep, tc.sleep, tc.expectedStatus, tc.expectedBody); err != nil {
+				t.Errorf("Failed request with initialSleep %v, sleep %v, with revision timeout %ds and expecting status %v: %v",
+					tc.initialSleep, tc.sleep, tc.timeoutSeconds, tc.expectedStatus, err)
+			}
+		})
+	}
+}
+
+func TestMaxDurationRevisionTimeout(t *testing.T) {
+	t.Parallel()
+	clients := test.Setup(t)
+	neverExpireMaxDurationSeconds := int64(100)
+	testCases := getDefaultTestCases()
+
+	// make sure high enough max duration has no effect in default cases
+	for _, tc := range testCases {
+		tc.maxDurationTimeoutSeconds = neverExpireMaxDurationSeconds
```

Comment:
fwiw I'd be tempted to have one single TestRevisionTimeouts testing all the timeout cases since they're all related anyway; extracting and modifying the base table to avoid copy pasting three test cases doesn't seem worth the extra boilerplate we end up with to me

Reply:
I thought about this but to me it seemed more readable to have the new tests cases separated, but merging all in one (initially had it like this) can be done.

Reply:
Merged the tests.